client/battle/CBattleInterface.h

```diff
@@ -198,16 +198,32 @@ class CBattleInterface : public CIntObject
 		SiegeHelper(const CGTownInstance * siegeTown, const CBattleInterface * _owner); //c-tor
 		~SiegeHelper(); //d-tor
 
-		//filename getters
-		//what: 0 - background,  1 - background wall,  2 - keep,          3 - bottom tower,  4 - bottom wall,
-		//      5 - below gate,  6 - over gate,        7 - upper wall,    8 - upper tower,   9 - gate,
-		//      10 - gate arch, 11 - bottom static    12 - upper static, 13 - moat,         14 - moat background,
-		//      15 - keep battlement, 16 - bottom battlement, 17 - upper battlement;
-		//      state uses EWallState enum
 		std::string getSiegeName(ui16 what) const;
-		std::string getSiegeName(ui16 what, int state) const;
-
-		void printPartOfWall(SDL_Surface * to, int what);//what: 1 - background wall, 2 - keep, 3 - bottom tower, 4 - bottom wall, 5 - below gate, 6 - over gate, 7 - upper wall, 8 - uppert tower, 9 - gate, 10 - gate arch, 11 - bottom static wall, 12 - upper static wall, 15 - keep creature cover, 16 - bottom turret creature cover, 17 - upper turret creature cover
+		std::string getSiegeName(ui16 what, int state) const; // state uses EWallState enum
+
+		void printPartOfWall(SDL_Surface * to, int what);
+
+		enum EWallVisual
+		{
+			BACKGROUND = 0,
+			BACKGROUND_WALL = 1,
+			KEEP,
+			BOTTOM_TOWER,
+			BOTTOM_WALL,
+			WALL_BELLOW_GATE,
+			WALL_OVER_GATE,
+			UPPER_WALL,
+			UPPER_TOWER,
+			GATE = 9,

```

Comment:
Yes. Enum like that is clearly much better than magic numbers with comments everywhere in code.